pkg/dataplane/http/context.go

```diff
@@ -308,54 +315,151 @@ func (c *context) GetItemsSync(getItemsInput *v3io.GetItemsInput) (*v3io.Respons
 		"PUT",
 		getItemsInput.Path,
 		"",
-		getItemsHeaders,
+		getItemsHeadersCapnp,
+//		getItemsHeaders,
 		marshalledBody,
 		false)
 
 	if err != nil {
 		return nil, err
 	}
+	contentType := string(response.HeaderPeek("Content-Type"))
 
-	c.logger.DebugWithCtx(getItemsInput.Ctx, "Body", "body", string(response.Body()))
+	if contentType != "application/octet-capnp" {
```

Comment:
Function is too long. Please split this into two helper functions - one parsing JSON and other capnp